handlers/issue.go

```diff
@@ -1,71 +1,83 @@
 package handlers
 
 import (
+	"context"
 	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
 	"regexp"
 	"strings"
 
 	"github.com/golang/glog"
 	"github.com/google/go-github/github"
 )
 
+const (
+	Assign   = "/assign"
+	Unassign = "/unassign"
+)
+
 type GithubIssue github.Issue
 
 func (s *Server) handleIssueEvent(body []byte) {
-	glog.Infof("Received an Issue Event")
+	glog.Info("Received an Issue Event")
 
 }
 
-func (s *Server) handleIssueCommentEvent(body []byte) {
-	glog.Infof("Received an IssueComment Event")
+//function to handle issue comments
+func (s *Server) handleIssueCommentEvent(body []byte, client *github.Client) {
+	var commentEvent github.IssueCommentEvent
 
-	var prc github.IssueCommentEvent
-	err := json.Unmarshal(body, &prc)
+	err := json.Unmarshal(body, &commentEvent)
 	if err != nil {
 		glog.Errorf("fail to unmarshal: %v", err)
 	}
-	glog.Infof("prc: %v", prc)
-	comment := *prc.Comment.Body
-
-	// https://github.com/islinwb/test/pull/1
-	prID := strings.SplitAfter(prc.Issue.PullRequestLinks.GetHTMLURL(), "github.com/")[1]
-	// https://github.com/islinwb/test/pull/1.patch
-	// From <commit ID> MON ...
-	patchURL := prc.Issue.PullRequestLinks.GetPatchURL()
-	resp, err := http.Get(patchURL)
-	if err != nil {
-		fmt.Println(err)
+	//Get the owner info from the req body, and use this owner for the all api's
+	owner := *commentEvent.Repo.Owner.Login
+	ctx := context.Background()
+	//extract the user from login to validate whether user is collaborator or not
+	user := commentEvent.Sender.Login
+	comment := *commentEvent.Comment.Body
+	//split the assignees and operation to be performed.
+	substrings := strings.Split(comment, "@")
+	// call to the function IsCollaborator checks whether the specified GitHub user has collaborator access or not
+	isCollaborator, _, err := client.Repositories.IsCollaborator(ctx, owner, *commentEvent.Repo.Name, *user)
+	if !isCollaborator {
+		glog.Errorf("Not the collaborator: %v", *user)
 	}
-
-	resp1, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		fmt.Println(err)
+		glog.Errorf("IsCollaborator returns error: %v", err)
 	}
-
-	patchDetail := string(resp1)
-	reg := regexp.MustCompile(`From [A-Za-z0-9]{40}`)
-	commitIDstr := reg.FindString(patchDetail)
-	commitID := strings.TrimPrefix(commitIDstr, "From ")
-
-	var info map[string]string
-	info["PR_ID"] = prID
-	info["Commit_ID"] = commitID
-
-	if labelReg.MatchString(comment) {
-		labelSlice := strings.Split(comment, " ")
-		if len(labelSlice) > 0 {
+	//regular expression to Assign or unassign the Assignees
+	reg := regexp.MustCompile("(?mi)^/(un)?assign(( @?[-\\w]+?)*)\\s*$")
+	matchAssignOrUnassign := reg.MatchString(comment)
+	//list of assignees to be assigned for issues/PR
+	listOfAssignees := make([]string, 0)
+	//range over the substring to get the list of assignees
+	for i, assignees := range substrings {
+		if i == 0 {
+			//first index is the operation to be performed,rest will be the assignees
```

Comment:
nit:
//first index is the operation to be performed, rest will be the assignees